packages/@lwc/engine/src/framework/api.ts

```diff
@@ -99,9 +99,18 @@ const CHAR_G = 103;
 const NamespaceAttributeForSVG = 'http://www.w3.org/2000/svg';
 const SymbolIterator = Symbol.iterator;
 
+function validateVNodeAfterElmCreation(vnode: VNode) {
+    if (process.env.NODE_ENV !== 'production') {
+        if (isFunction(vnode.data.validateFn)) {
+            vnode.data.validateFn.call(vnode);
+        }
+    }
+}
+
 const TextHook: Hooks = {
     create: (vnode: VNode) => {
         vnode.elm = document.createTextNode(vnode.text as string);
+        validateVNodeAfterElmCreation(vnode);
```

Comment:
if we can agree on the validation to only be used in dev mode, then we can probably wrap this function call with the if condition for that.